src/api/controllers/ItemImageController.ts

```diff
@@ -29,14 +37,88 @@ export class ItemImageController {
+    @httpPost('/template/:templateId')
+    public async create( @response() res: myExpress.Response, @requestParam('templateId') templateIdStr: string,
+                         @requestBody() body: any, @request() req: any): Promise<any> {
+        // check templateId is number
+        // also check listingItemTemplate id present in params
+        let templateId: number;
+        if (!templateIdStr) {
+            throw new MessageException('ListingItemTemplate id can not be null.');
+        } else {
+            templateId = parseInt(templateIdStr, 10);
+            if ( !templateId ) {
+                throw new MessageException('ListingItemTemplate id must be an integer.');
+            }
+        }
+
+        // this.log.debug('templateId: ' + templateId);
+        // this.log.debug('files: ', req.files);
+
+        for ( const file of req.files ) {
+            // this.log.debug(`Found image <${file.fieldname}>.`);
+            if ( file.fieldname === 'image' ) {
+                const imageFile = file;
+
+                // Read the file data in
+                const dataStr = fs.readFileSync(imageFile.path, 'base64');
+                // this.log.error('dataStr = ' + dataStr);
+
+                // find listing item template
+                const listingItemTemplate = await this.listingItemTemplateService.findOne(templateId);
+                this.log.debug('imageFile.mimetype = ' + imageFile.mimetype);
+                // find related itemInformation
+                const itemInformation = await listingItemTemplate.related('ItemInformation').toJSON();
+                const itemImage = itemInformation.ItemImages[0];
+                if ( !itemImage ) {
+                    // Doesn't exist yet, create instead of update
+                    const tmpDataId = this.makeid();
+                    return await this.itemImageService.create(
+                        {
+                            item_information_id: itemInformation.id,
+                            hash: ObjectHash.getHash(itemInformation),
+                            data: {
+                                protocol: ImageDataProtocolType.LOCAL,
+                                encoding: 'BASE64',
+                                data: dataStr,
+                                dataId: tmpDataId,
+                                originalMime: imageFile.mimetype,
+                                originalName: imageFile.originalname
+                            }
+                        } as ItemImageUpdateRequest);
+                } else {
+                    const itemImageId = itemImage.id;
+                    // this.log.debug('itemImage = ' + JSON.stringify(itemImage));
+                    this.log.debug('itemImageId = ' + itemImageId);
+
+                    // create item images
+                    return await this.itemImageService.update(
+                        itemImageId,
```

Comment:
if item allready has one image then it's updated, why?

Reply:
I don't understand.
Do you want it so if an image already exists we just disregard the upload and use what we already have?